codecs/yuv/src/bt601.rs

```diff
@@ -0,0 +1,122 @@
+//! YUV-to-RGB decode
+
+fn clamp(v: f32) -> u8 {
+    if v.is_nan() {
+        return 0;
+    }
+
+    if v < 0.0 {
+        return 0;
+    }
+
+    if v > 255.0 {
+        return 255;
```

Comment:
After saturating conversions became default in Rust, `clamp` could be simply `(v + 0.5) as u8`, right?

Reply:
Hah, good point! IMHO, yes.

Reply:
While nice and concise, I find ithis to be a lot slower. No idea why.

Reply:
Oh, sorry, I compared it against an already modified local version: `(v + 0.5).max(0.0).min(255.0) as u8`

Reply:
Maybe these explicit `.min().max()` calls are not as careful/complete as the one that comes with the cast (when used on its own), but they are enough for the compiler to drop these checks from the cast?

Reply:
I think I got lost, not sure which you mean is faster in the end :)
But lyon author had a similar observation that `.max()` is sometimes slower than expected:
https://github.com/nical/lyon/blob/4c31d8c9907f03ad5962d02e22e2783bfdfbfdb0/tessellation/src/fill.rs#L28

Reply:
https://godbolt.org/z/jdYT48

Reply:
See: https://github.com/kmeisthax/ruffle/pull/12

Reply:
With everything else being equal, measuring only the difference between the three clamp variants in the godbolt you linked, the same way as in my PR (on top of the changes of that PR):
![image](https://user-images.githubusercontent.com/288816/105246066-051bb280-5b73-11eb-9861-df1e6b5d0409.png)